src/FormField/ScopeBuilder.php

```diff
@@ -0,0 +1,565 @@
+    }
+
+    protected static function getRule($type, array $defaults = [], Field $field = null): array
+    {
+        $rule = self::$ruleTypes[strtolower($type)] ?? self::$ruleTypes['default'];
+
+        // when $rule is an alias
+        if (is_string($rule)) {
+            return self::getRule($rule, $defaults, $field);
+        }
+
+        $options = $defaults['options'] ?? [];
+
+        // 'options' is atk specific so not necessary to pass it to VueQueryBuilder
+        unset($defaults['options']);
+
+        // when $rule is callable
+        if (is_callable($rule)) {
+            $rule = call_user_func($rule, $field, $options);
+        }
+
+        // map all values for callables and merge with defaults
+        return array_merge(array_map(function ($value) use ($field, $options) {
+            return is_array($value) && is_callable($value) ? call_user_func($value, $field, $options) : $value;
+        }, $rule), $defaults);
+    }
+
+    /**
+     * Returns the choises array for the field rule.
+     */
+    protected static function getChoices(Field $field, $options = []): array
+    {
+        $choices = [];
+        if ($field->enum) {
+            $choices = array_combine($field->enum, $field->enum);
+        }
+        if ($field->values && is_array($field->values)) {
+            $choices = $field->values;
+        } elseif ($field->reference) {
+            $model = $field->reference->refModel();
+
+            if ($limit = $options['limit'] ?? false) {
+                $model->setLimit($limit);
+            }
+
+            foreach ($model as $item) {
+                $choices[$item[$model->id_field]] = $item[$model->title_field];
+            }
+        }
+
+        $ret = [['label' => '[empty]', 'value' => null]];
+        foreach ($choices as $value => $label) {
+            $ret[] = compact('label', 'value');
+        }
+
+        return $ret;
+    }
+
+    public function renderView()
+    {
+        $this->app->addStyle('
+            .vue-query-builder select,input {
+                width: auto !important;
+            }
+        ');
+
+        $this->scopeBuilderView->template->trySetHTML('Input', $this->getInput());
+
+        parent::renderView();
+
+        $this->scopeBuilderView->vue(
+            'atk-query-builder',
+            [
+                'rules' => $this->rules,
+                'maxDepth' => $this->maxDepth,
+                'query' => $this->query,
+            ]
+        );
+    }
+
+    /**
+     * Converts an VueQueryBuilder query array to Condition or Scope.
+     */
+    public static function queryToScope(array $query): AbstractScope
+    {
+        $type = $query['type'] ?? 'query-builder-group';
+        $query = $query['query'] ?? $query;
+
+        switch ($type) {
+            case 'query-builder-group':
+                $components = array_map([static::class, 'queryToScope'], $query['children']);
+                $junction = $query['logicalOperator'] === 'all' ? Scope::AND : Scope::OR;
```

Comment:
I do not like `all` name

Reply:
![image](https://user-images.githubusercontent.com/2228672/84361748-65ee1000-abcc-11ea-9f88-0f036e4c6152.png)

In GUI we should have definitely `OR` and/or `AND` - people are used to use these even from non-SQL world

Reply:
should be also updated in data/scope: `scope->all` is simply not ok

Reply:
Changed